### PR TITLE
AccessControl: SQL filters for team search

### PR DESCRIPTION
**Pull request by gamab**
**What this PR does / why we need it**:
This PR brings accesscontrol to the GET endpoints for teams. It also filters out teams at SQL level for the search endpoint.

For example, if we have 5 teams:
image

And allow members of the "team admins" teams to see three teams only:
```yaml
  - name: "custom:teams:123:writer"
    description: "Read and Write a subset (id: 1, 2, 3) of teams"
    version: 1
    global: true
    permissions:
      - action: "teams:read"
        scope: "teams:id:1"
      - action: "teams:write"
        scope: "teams:id:1"
      - action: "teams.permissions:read"
        scope: "teams:id:1"
      - action: "teams.permissions:write"
        scope: "teams:id:1"
      - action: "teams:read"
        scope: "teams:id:2"
      - action: "teams:write"
        scope: "teams:id:2"
      - action: "teams.permissions:read"
        scope: "teams:id:2"
      - action: "teams.permissions:write"
        scope: "teams:id:2"
      - action: "teams:read"
        scope: "teams:id:3"
      - action: "teams:write"
        scope: "teams:id:3"
      - action: "teams.permissions:read"
        scope: "teams:id:3"
      - action: "teams.permissions:write"
        scope: "teams:id:3"
    teams:
      - name: "team admins"
        global: true
```

The end result for a member of the "team admins" team will be:
image

Thus filtering out the teams "hidden to others" and "hidden too".

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana-enterprise/issues/2605
Fixes https://github.com/grafana/grafana-enterprise/issues/2668

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/services/sqlstore/team.go
Comment (kalleep) [inline pkg/services/sqlstore/team.go:265]: nice that you covered count as well!

**Review by kalleep — APPROVED** [review]

**Review by Jguer — CHANGES_REQUESTED** [review]
Blocking changes with upstream epic

**Review by IevaVasiljeva — APPROVED** [review]
Looks good to go once we're ready to merge. Great work!

**Review by Jguer — APPROVED** [review]

**Pull request merged**
